OHIF/Viewers
# feat(cloud data source config): GUI and API for configuring a cloud data source with Google cloud healthcare implementation

### Context

OHIF requires a GUI interface to configure a cloud server API data source such as...
- Google cloud healthcare
- AWS health imaging
- Azure

This is a PR for configuring a Google dicom web data source.



### Changes & Results

Introduced generic interfaces (i.e. BaseDataSourceConfigurationAPI, BaseDataSourceConfigurationAPIItem)
to facilitate the configuration of an arbitrary data source - in particular
a cloud data source such as Google.

Implemented a generic GUI that uses the generic interfaces to allow users to configure a data source
via a GUI.

Implemented concrete implementations of the interfaces for Google.

Introduced the InputFilterText GUI component to handle the input of filtering text.



### Testing

1. Checkout the `jbocce:feat/cloud-server-api-gui-picker` branch.
2. Set the APP_CONFIG environment variable to `'config/google.js'`.
3. `yarn install`
4. `yarn start`
5. Click on the Google data source summary and/or gear icon next to the `Study List` label.
6. Configure various Google dicom stores. Those most useful are under:  `OHIF Cloud Healthcare - us-east4` and `OHIF Cloud Healthcare - us-central1`.

The various GUI components, modals, panels, etc. should closely approximate what was provided by @dan-rukas below. Thanks Dan!

![image](https://github.com/OHIF/Viewers/assets/109477394/783edb65-1929-4282-a39c-ebe170867505)


![image](https://github.com/OHIF/Viewers/assets/109477394/ef97ed50-5fc9-41a6-9e92-40efcba1ef79)



### Checklist

#### PR



  semantic-release format and guidelines.

#### Code

  etc.)

#### Public Documentation Updates



  additions or removals.

#### Tested Environment

- [x] Browser: Chrome 115.0.5790.171
  


[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch
